### PR TITLE
Blaze: Fix a crash when no ad suggestions

### PR DESCRIPTION
**Pull request by 0nko**
This PR fixes a crash when there are no ad suggestions in the ad-edit screen.

**To test:**

Online
1. Start a campaign creation
2. Tap on the Edit ad button
3. Notice that 3 suggestions are loaded
4. Edit something and tap save
5. Notice the ad details are updated
6. Tap on the Edit ad button again
7. Notice the correct details are loaded
8. Tap on the arrows and notice the other 3 suggestions are available

Offline
1. Turn on the flight mode
2. Start a campaign creation
3. Tap on the Edit ad button
4. Notice the AI suggestions row is not displayed
5. Add suggestion details and save
6. Notice the details are updated
7. Tap on the Edit ad button again
8. Notice the correct details are loaded

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/193">17.4 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by JorgeMucientes** [comment]
Hey @0nko I think I fixed this already as part of this PR:https://github.com/woocommerce/woocommerce-android/pull/10820
Specifically in this commit: https://github.com/woocommerce/woocommerce-android/pull/10820/commits/c726565a6458a988c0eaab9f7a41191c23d66feb, where AI suggestions are passed as `navArgs` from CampaignPreviewFragment to EditAdFragment. The reason for this where to avoid doing unnecessary requests. 

So, I'm unsure this is needed. I tested following the steps shared in the description and the app doesn't crash with this PR's code https://github.com/woocommerce/woocommerce-android/pull/10820

Let me know if I missed something and I can address it :)

**Comment by hichamboushaba** [comment]
@JorgeMucientes did you test the offline case, because I'm not sure your changes will fix this case though, as the passed suggestions will still be empty when we get an error. But TBH, I didn't review the whole code yet, just wondering.

**Comment by JorgeMucientes** [comment]
Hey @hichamboushaba resuming this discussion :) 

> did you test the offline case, because I'm not sure your changes will fix this case though, as the passed suggestions will still be empty when we get an error. But TBH, I didn't review the whole code yet, just wondering.

Yes, [the changes](https://github.com/woocommerce/woocommerce-android/commit/73d47ba3a070f0a7cf19f244b58f04743226d414) added there fix the crash. It's not really about passing the suggestions from the preview screen to the edit ad screen; that fixed the problem. It was a small refactor done on how the suggestion list is processed that fixed it.

The following screencast was recorded from this PR: #10926

https://github.com/woocommerce/woocommerce-android/assets/2663464/62e2894d-ff08-4101-ba36-583d5d73fd9f

@0nko I think we can discard the changes added to `BlazeCampaignCreationEditAdViewModel` and keep only the changes applied to the screen UI code to hide the suggestions row if the ai suggestion list is empty.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/196">17.6</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by 0nko** [comment]
@JorgeMucientes I've updated the code, ready for a review.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/ad/BlazeCampaignCreationEditAdViewModel.kt [hunk 1]
```diff
@@ -47,40 +47,30 @@ class BlazeCampaignCreationEditAdViewModel @Inject constructor(
...
-                        suggestionIndex = 0
-                    )
-                }
+            val passedDetails = AiSuggestionForAd(navArgs.tagline, navArgs.description)
+            val suggestions = navArgs.aiSuggestionsForAd.toList()
+            _viewState.update {
+                it.copy(
+                    suggestions = listOf(passedDetails) + (suggestions - passedDetails),
+                    suggestionIndex = 0
+                )
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/ad/BlazeCampaignCreationEditAdViewModel.kt:56]: Nice simplification!! 👍🏼

**Review by JorgeMucientes — APPROVED** [review]

**Comment by JorgeMucientes** [comment]
The code looks great, and everything works as expected! Thanks @0nko 👍🏼

**Pull request merged**